File: count_possible_ways_construct_buildings.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
 public:
  int TotalWays(int N) {
    // Code here
    int const mod = 1000000007;
    vector<long long> dp(N + 1);
    dp[0] = 1;
    dp[1] = 2;
    for (int i = 2; i <= N; ++i) {
      dp[i] = (dp[i - 2] + dp[i - 1]) % mod;
    }
    return (dp[N]) * (dp[N]) % mod;
  }
};

//{ Driver Code Starts.
int main() {
  int tc;
  cin >> tc;
  while (tc--) {
    int N;
    cin >> N;
    Solution ob;
    int ans = ob.TotalWays(N);
    cout << ans << "\n";
  }
  return 0;
}
// } Driver Code End

File: dp/count_possible_ways_construct_buildings.py
# User function Template for python3


class Solution:
    def TotalWays(self, N):
        # Code here
        dp = [0] * (N + 1)
        M = int(1e9 + 7)
        dp[0] = 1
        dp[1] = 2
        for i in range(2, N + 1):
            dp[i] = (dp[i - 1] + dp[i - 2]) % M

        return dp[N] * dp[N] % M


# {
# Driver Code Starts
# Initial Template for Python 3


if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        N = int(input())
        ob = Solution()
        ans = ob.TotalWays(N)
        print(ans)
# } Driver Code Ends
